tools/publish_aws.py

```diff
@@ -143,6 +120,38 @@ def print_help(argv):
     logger.info('In addition, environment variables named \'TEMPLATE_SOME_PARAMETER\' will be inserted against the provided package template (with params of the form \'{{some-parameter}}\')')
 
 
+def s3_urls_from_env(package_name):
+    s3_bucket = os.environ.get("S3_BUCKET") or "infinity-artifacts"
+    logger.info("Using artifact bucket: {}".format(s3_bucket))
+
+    s3_dir_path = os.environ.get("S3_DIR_PATH") or "autodelete7d"
+    s3_dir_name = os.environ.get("S3_DIR_NAME")
+    if not s3_dir_name:
+        s3_dir_name = "{}-{}".format(
```

Comment:
Not specific to this PR, but strftime will only parse a string into time when the string is in the specific format. While this might always be the case here, it's a bit limiting. dateutil.parse will convert any string to a python datetime object.